orderer/common/channelparticipation/restapi.go

```
@@ -289,6 +290,8 @@ func (h *HTTPHandler) serveRemove(resp http.ResponseWriter, req *http.Request) {
 		h.sendResponseNotAllowed(resp, errors.WithMessage(err, "cannot remove"), http.MethodGet)
 	case types.ErrChannelNotExist:
 		h.sendResponseJsonError(resp, http.StatusNotFound, errors.WithMessage(err, "cannot remove"))
+	case types.ErrChannelPendingRemoval, types.ErrChannelRemoval:
+		h.sendResponseJsonError(resp, http.StatusConflict, errors.WithMessage(err, "cannot remove"))
```

`types.ErrChannelRemoval` - I suggest `types.ErrChannelRemovalFailure`, to be more explicit.
In addition, I think it should return the error code "InternalServerError" 500. What do you think?

---

Yes I think that's better. 🧹